### PR TITLE
documentation improvements to match ui toolkit v0.9.0

### PR DESCRIPTION
**Pull request by ikalafat**
documentation improvements to match ui toolkit v0.9.0

- added richmedia component docs
- added video component docs
- added lightbox component docs
- improved grid view (gridrow) component and added larger example

### CODE REVIEW

File: _layouts/doc.html [hunk 1]
```diff
@@ -86,6 +85,9 @@
 							<li><a href="{{ site.baseurl }}/docs/ui-toolkit/components/headers">Headers</a></li>
 							<li><a href="{{ site.baseurl }}/docs/ui-toolkit/components/overlay">Overlay</a></li>
 							<li><a href="{{ site.baseurl }}/docs/ui-toolkit/components/text-input">TextInput</a></li>
+							<li><a href="{{ site.baseurl }}/docs/ui-toolkit/components/video">Video</a></li>
+							<li><a href="{{ site.baseurl }}/docs/ui-toolkit/components/lightbox">Lightbox</a></li>
+							<li><a href="{{ site.baseurl }}/docs/ui-toolkit/components/rich-media">Rich Media</a></li>
```
Comment (ikalafat) [inline _layouts/doc.html]: Done. Updated `NavigationBar`, `DropDownMenu`, `ListView` and `GridView` to be without space, to be consistent with official RN docs.

File: _posts/1-2-15-RichMedia.md [hunk 2]
```diff
@@ -7,61 +7,42 @@ section: UI toolkit
...
-> davor: we need more explanation here like: Attachments are Shoutem specific DOM elements for video and image galleries that can be insterted within the HTML.  
-> kale: rephrased a bit, see comment below  
-> davor: Why are we not passing attachments within HTML?  
-> kale: check this https://fiveminutes.jira.com/wiki/display/SE/Data+Exchange+Protocol  
-> kale: DEP is designed to reference attachments by using `<attachment>` tags in body, and then to pull those attachments from `attachment` JSON object
-
 * **body** : string  
-  - Prop containing HTML code that will be rendered using this Component  
+  - Body of the HTML content that will be transformed by this component and rendered using React Native and Shoutem UI toolkit components.  
+* **onError** : func
```
Comment (ikalafat) [inline _posts/1-2-15-RichMedia.md]: Fixed. Thanks.

File: _posts/1-2-15-RichMedia.md [hunk 3]
```diff
@@ -7,61 +7,42 @@ section: UI toolkit
...
+  - Callback function triggered if RichMedia component fails to parse given HTML (in `body` prop).
+* **openURL(url)** : func
```
Comment (ikalafat) [inline _posts/1-2-15-RichMedia.md]: Fixed, thanks.

File: _posts/1-2-17-Video.md [hunk 4]
```diff
@@ -8,8 +8,7 @@ section: UI toolkit
 # Video
 
 Video component is a component that can be used to render all types of Video items.  
-- If the source is Youtube or Vimeo, then the `NativeVideo` component plays the video  
-- In all other cases, the video is displayed in a WebView using `WebViewVideo` component  
+It renders a Video based on the source type. If source is an url to a web player the video is displayed in a WebView, if not, then a Video HTML element is displayed in the WebView.
```
Comment (ikalafat) [inline _posts/1-2-17-Video.md]: Should be good now.

File: _posts/1-2-27-Lightbox.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+---
+layout: doc
+permalink: /docs/ui-toolkit/components/lightbox
+title: Lightbox
+section: UI toolkit
+---
+
+# Lightbox
+
+This Lightbox is Shoutem's fork of [`react-native-lightbox`](https://github.com/oblador/react-native-lightbox) component.
```
Comment (ikalafat) [inline _posts/1-2-27-Lightbox.md]: Done

File: _posts/1-2-27-Lightbox.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+layout: doc
+permalink: /docs/ui-toolkit/components/lightbox
+title: Lightbox
+section: UI toolkit
+---
+
+# Lightbox
+
+This Lightbox is Shoutem's fork of [`react-native-lightbox`](https://github.com/oblador/react-native-lightbox) component.  
+All content within Lightbox is transformable, meaning it can be pinched to zoom, panned (translated) etc.
```
Comment (ikalafat) [inline _posts/1-2-27-Lightbox.md]: Should be good now.

File: _posts/1-2-27-Lightbox.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    style={% raw %}{{ height: 300 }}{% endraw %}
+    source={% raw %}{{ uri: 'http://knittingisawesome.com/wp-content/uploads/2012/12/cat-wearing-a-reindeer-hat1.jpg' }}{% endraw %}
+  />
+</Lightbox>
+```
+
+#### Props
+
+* **activeProps** : object  
+  - Optional set of props applied to the content component when in lightbox mode. Usable for applying custom styles or higher resolution image source.
```
Comment (ikalafat) [inline _posts/1-2-27-Lightbox.md]: Done

File: _posts/1-2-17-Video.md [hunk 8]
```diff
@@ -8,8 +8,7 @@ section: UI toolkit
 # Video
 
 Video component is a component that can be used to render all types of Video items.
```
Comment (ikalafat) [inline _posts/1-2-17-Video.md]: should be good.

File: _posts/1-2-27-Lightbox.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+* **renderHeader(close)** : function
+  - Callback function that should return custom header instead of default with X button in upper left corner. 
+* **renderContent** : function
+  - Callback function that should return custom Lightbox content instead of default child content.
+* **onClose**: function
+  - Callback function triggered when Lightbox is closed.
+* **onOpen**: function
+  - Callback function triggered when Lightbox is opened.
```
Comment (ikalafat) [inline _posts/1-2-27-Lightbox.md]: removed `callback function` where it wasn't necessary and rephrased a bit. also marked components. should be good now.

**Comment by ikalafat** [comment]
@tenodi  please take another look.

File: _layouts/doc.html [hunk 10]
```diff
@@ -64,18 +64,17 @@
 						<a class="menu-group-title" href="{{ site.baseurl }}/docs/ui-toolkit/components/typography">Components</a>
 						<ul id="ui-toolkit-components" class="menu-group-wrapper level-3">
 							<li><a href="{{ site.baseurl }}/docs/ui-toolkit/components/typography">Typography</a></li>
-							<li><a href="{{ site.baseurl }}/docs/ui-toolkit/components/navigation-bar">Navigation Bar</a></li>
-							<li><a href="{{ site.baseurl }}/docs/ui-toolkit/components/dropdown-menu">Dropdown Menu</a></li>
-							<li><a href="{{ site.baseurl }}/docs/ui-toolkit/components/list-view">List View</a></li>
-							<li><a href="{{ site.baseurl }}/docs/ui-toolkit/components/grid-view">Grid View</a></li>
+							<li><a href="{{ site.baseurl }}/docs/ui-toolkit/components/navigation-bar">NavigationBar</a></li>
+							<li><a href="{{ site.baseurl }}/docs/ui-toolkit/components/dropdown-menu">DropdownMenu</a></li>
```
Comment (tenodi) [inline _layouts/doc.html]: If you decided to remove the space, components need to be named the same as [here](https://github.com/shoutem/ui/blob/develop/components/DropDownMenu.js).

File: _posts/1-2-17-Video.md [hunk 11]
```diff
@@ -7,9 +7,8 @@ section: UI toolkit
 
 # Video
 
-Video component is a component that can be used to render all types of Video items.  
-- If the source is Youtube or Vimeo, then the `NativeVideo` component plays the video  
-- In all other cases, the video is displayed in a WebView using `WebViewVideo` component  
+`Video` component can be used to render all types of Video items.  
+It renders a Video based on the source type. If source is an url to a web player the video is displayed in a `WebView`, if not, then a `Video` `HTML` element is displayed in the `WebView`.
```
Comment (tenodi) [inline _posts/1-2-17-Video.md]: Video can be with small starting letter

Comment (tenodi) [inline _posts/1-2-17-Video.md]: I think this is a better form: If a source is `URL` to a web player, video is displayed in a `WebView`. If not, `video` `HTML` element is displayed in the `WebView`.

File: _posts/1-2-27-Lightbox.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+layout: doc
+permalink: /docs/ui-toolkit/components/lightbox
+title: Lightbox
+section: UI toolkit
+---
+
+# Lightbox
+
+`Lightbox` is Shoutem's fork of [`react-native-lightbox`](https://github.com/oblador/react-native-lightbox) component.  
+All content within `Lightbox` is transformable, meaning it can be pinched to zoom, panned, (translated) etc.
```
Comment (tenodi) [inline _posts/1-2-27-Lightbox.md]: Shouldn't comma come after `(translated`) 😛?

File: _posts/1-2-9-DropDownMenu.md [hunk 13]
```diff
@@ -1,7 +1,7 @@
 ---
 layout: doc
 permalink: /docs/ui-toolkit/components/dropdown-menu
-title: Dropdown Menu
+title: DropdownMenu
```
Comment (tenodi) [inline _posts/1-2-9-DropDownMenu.md]: Same comment as in site's navigation.

File: _layouts/doc.html (hunk 10 above)
Comment (ikalafat) [inline _layouts/doc.html]: my bad. fixed

File: _posts/1-2-17-Video.md (hunk 11 above)
Comment (ikalafat) [inline _posts/1-2-17-Video.md]: Fixed, changed text to your proposal

File: _posts/1-2-27-Lightbox.md (hunk 12 above)
Comment (ikalafat) [inline _posts/1-2-27-Lightbox.md]: Well, you sad after panned in your previous comment :P

File: _posts/1-2-9-DropDownMenu.md (hunk 13 above)
Comment (ikalafat) [inline _posts/1-2-9-DropDownMenu.md]: fixed

**Comment by tenodi** [comment]
Looks good!

**Pull request merged**